subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -208,6 +215,25 @@ static int update_job_execution(struct mqtt_client *const client,
 	return 0;
 }
 
+/**
+ * @brief Update the job document of the current job to AWS_JOBS_FAILED and move to ERROR state.
+ *
+ * @param client Connected MQTT client instance
+ * @return 0 If successful otherwise a negative error code is returned.
+ */
+static int set_current_job_failed(struct mqtt_client *const client)
+{
+	struct aws_fota_event aws_fota_evt;
+
+	aws_fota_evt.id = AWS_FOTA_EVT_ERROR;
+	callback(&aws_fota_evt);
+	internal_state_set(STATE_ERROR);
+	return update_job_execution(client,
+				    job_id_handling,
+				    sizeof(job_id_handling),
```

Comment:
Since we are not actually using the buffer length for anything except for a string comparison and work with the null termination internally anyways, I would actually propose getting rid of this parameter and just using `sizeof(job_id_handling)` directly in the one `strncmp` in `update_job_execution()`.